File: 백준/Silver/10844. 쉬운 계단 수/쉬운 계단 수.java
import java.io.*;
import java.util.Arrays;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        int n = Integer.parseInt(br.readLine());
        int[][] d = new int[n][10];
        int mod = 1_000_000_000;

        Arrays.fill(d[0], 1);
        d[0][0] = 0;
        for (int i=1; i<n; i++) {
            d[i][0] = d[i-1][1] % mod;
            for (int j=1; j<9; j++)
                d[i][j] = (d[i-1][j-1] + d[i-1][j+1]) % mod;
            d[i][9] = d[i-1][8] % mod;
        }

        int sum = 0;
        for (int i: d[n-1])
            sum = (sum + i) % mod;
        System.out.print(sum);
    }
}

File: 백준/Silver/10844. 쉬운 계단 수/쉬운 계단 수.py
n = int(input())
d = [[0] * 10 for _ in range(n+1)]
mod = 1_000_000_000
for i in range(1, 10) :
    d[1][i] = 1
for i in range(2, n+1) :
    d[i][0] = d[i-1][1]
    for j in range(1, 9) :
        d[i][j] = (d[i-1][j-1] + d[i-1][j+1]) % mod
    d[i][9] = d[i-1][8]
print(sum(d[n]) % mod)